hudi-utilities/src/main/java/org/apache/hudi/utilities/schema/HoodieSparkSchemaProvider.java

```diff
@@ -18,41 +18,27 @@
 
 package org.apache.hudi.utilities.schema;
 
-import org.apache.hudi.ApiMaturityLevel;
-import org.apache.hudi.PublicAPIClass;
-import org.apache.hudi.PublicAPIMethod;
 import org.apache.hudi.common.config.TypedProperties;
+import org.apache.hudi.schema.SchemaProvider;
 
-import org.apache.avro.Schema;
 import org.apache.spark.api.java.JavaSparkContext;
 
-import java.io.Serializable;
-
 /**
- * Class to provide schema for reading data and also writing into a Hoodie table.
+ * SchemaProvider for spark engine.
  */
-@PublicAPIClass(maturity = ApiMaturityLevel.STABLE)
-public abstract class SchemaProvider implements Serializable {
+public abstract class HoodieSparkSchemaProvider extends SchemaProvider {
```

Comment:
IMO, can we make SchemaProvider spark-free so that we do not need to copy a FilebasedSchemaProvider for Flink. wdyt?

Reply:
> IMO, can we make SchemaProvider spark-free so that we do not need to copy a FilebasedSchemaProvider for Flink. wdyt?

Good idea, I'll give it a try